azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -15,34 +15,389 @@
  */
 package azkaban.executor.container;
 
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.Constants.ContainerizedExecutionManagerProperties;
+import azkaban.container.models.AzKubernetesV1PodBuilder;
+import azkaban.container.models.AzKubernetesV1SpecBuilder;
+import azkaban.container.models.ImagePullPolicy;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutableFlowBase;
+import azkaban.executor.ExecutableNode;
 import azkaban.executor.ExecutorLoader;
 import azkaban.executor.ExecutorManagerException;
 import azkaban.utils.Props;
+import com.google.common.collect.ImmutableMap;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1DeleteOptions;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodSpec;
+import io.kubernetes.client.openapi.models.V1Status;
+import io.kubernetes.client.util.ClientBuilder;
+import io.kubernetes.client.util.KubeConfig;
+import io.kubernetes.client.util.Yaml;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
 import javax.inject.Inject;
+import javax.inject.Singleton;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * TODO: Add implementation for integration with Kubernetes once it is tested on cluster.
+ * This class is Kubernetes based implementation for containerization. It has implementation for
+ * creation/deletion of Pod and service. For any execution, it will identify version set and create
+ * a pod for all the valid jobTypes of a flow.
  */
+@Singleton
 public class KubernetesContainerizedImpl implements ContainerizedImpl {
+
+  public static final String DEFAULT_FLOW_CONTAINER_NAME_PREFIX = "az-flow-container";
+  public static final String DEFAULT_POD_NAME_PREFIX = "fc-dep";
+  public static final String DEFAULT_SERVICE_NAME_PREFIX = "fc-svc";
+  public static final String DEFAULT_CLUSTER_NAME = "azkaban";
+  public static final String DEFAULT_CPU_LIMIT = "4";
```

Comment:
4 is too much as default. We have a max of 56 cores available. We will not be able to schedule too many pods on a node with that as default. I think the default should be 1.

Reply:
Good point. I will set both limit and request as 1 for default. I will add configuration for these parameters to add an option to take it from property.